chapter9/phone-agent/webrtc_app.py

```
@@ -0,0 +1,411 @@
+    if not offer.startswith("v=0") or len(offer) > 1_000_000:
+        raise HTTPException(status_code=400, detail="invalid SDP offer")
+
+    pc = RTCPeerConnection()
+    PEERS[call_id] = pc
+    pc.addTrack(ConnectionAudioTrack())
+
+    @pc.on("track")
+    def on_track(track: Any) -> None:
+        if track.kind == "audio":
+            record["transport"]["local_audio_track"] = True
+            asyncio.create_task(_consume_microphone(track, record))
```

Ruff (0.16.0)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)